cluster-autoscaler/core/scaledown/planner/planner.go

```diff
@@ -84,10 +85,12 @@ func New(context *context.AutoscalingContext, processors *processors.Autoscaling
 func (p *Planner) UpdateClusterState(podDestinations, scaleDownCandidates []*apiv1.Node, as scaledown.ActuationStatus, pdb []*policyv1.PodDisruptionBudget, currentTime time.Time) errors.AutoscalerError {
 	p.latestUpdate = currentTime
 	p.actuationStatus = as
-	// TODO: clone cluster snapshot to avoid persisting changes done by the
-	// simulation. Or - better yet - allow the snapshot to be forked twice
-	// and just fork it here.
-	err := p.injectOngoingActuation()
+	err := p.rs.ForkClusterSnapshot()
+	if err != nil {
```

Comment:
Forking the snapshot shouldn't return an error anymore. If forking can happen multiple times, `Fork()` shouldn't return a value at all. This will simplify error handling in a few places.

Reply:
>Forking the snapshot shouldn't return an error anymore
Why? I added a fork limit, so the Fork() can return error.

Reply:
Done